packages/amplify_api_plugin_interface/lib/src/GraphQL/GraphQLRequest.dart

```diff
@@ -18,22 +18,34 @@ import 'package:flutter/foundation.dart';
 import '../UUID.dart';
 
 class GraphQLRequest<T> {
+  String apiName;
   String document;
   Map<String, dynamic> variables = {};
   String cancelToken = UUID.getUUID();
 
-  GraphQLRequest({@required this.document, Map<String, dynamic> variables}) {
+  GraphQLRequest(
+      {String apiName,
+      @required this.document,
+      Map<String, dynamic> variables}) {
     if (variables != null) {
       this.variables = variables;
     }
+
+    if (apiName != null) {
+      this.apiName = apiName;
+    }
   }
 
   Map<String, dynamic> serializeAsMap() {
     final Map<String, dynamic> result = <String, dynamic>{};
-
     result['document'] = this.document;
     result['variables'] = this.variables;
     result['cancelToken'] = this.cancelToken;
+
+    if (this.apiName != null) {
```

Comment:
I think the null check is necessary because the apiName is an optional value, so if the value is null we want to avoid set it in the serialized map. If not, I experienced an unexpected behavior in the native platforms.

Reply:
Yes, I believe you are correct as I was able to cause an ios error when I removed this check. thanks.